scripts/all-tests.sh

```diff
@@ -95,126 +95,69 @@ END
         exit $EXIT
         ;;
 
-    # TODO: Remove the "unit" TEST_SUITE in favor of "lms-unit", etc.
-    # For now it is left in here so that there is not a time limit on how fast
-    # we need to update the groovy script in the jenkins config after merging
-    # the changes needed to shard out the unit tests.
-    "unit")
-        case "$SHARD" in
-            "lms")
-                SHARD=1 paver test_system -s lms --extra_args="--with-flaky" --cov_args="-p" || { EXIT=1; }
-                ;;
-            "cms-js-commonlib")
-                SHARD=1 paver test_system -s cms --extra_args="--with-flaky" --cov_args="-p" || { EXIT=1; }
-                SHARD=1 paver test_js --coverage --skip_clean || { EXIT=1; }
-                SHARD=1 paver test_lib --skip_clean --extra_args="--with-flaky" --cov_args="-p" || { EXIT=1; }
-                ;;
-            *)
-                paver test --extra_args="--with-flaky"
-                paver coverage
-                ;;
-        esac
-        exit $EXIT
-        ;;
-
     "lms-unit")
         case "$SHARD" in
             "1")
-                paver test_system -s lms --extra_args="--attr='shard_1' --with-flaky" --cov_args="-p" || { EXIT=1; }
+                paver test_system -s lms --extra_args="--attr='shard_1' --with-flaky" --cov_args="-p"
                 ;;
             "2")
-                paver test_system -s lms --extra_args="--attr='shard_1=False' --with-flaky" --cov_args="-p" || { EXIT=1; }
+                paver test_system -s lms --extra_args="--attr='shard_1=False' --with-flaky" --cov_args="-p"
                 ;;
             *)
-                paver test_system -s lms --extra_args="--with-flaky" --cov_args="-p" || { EXIT=1; }
+                paver test_system -s lms --extra_args="--with-flaky" --cov_args="-p"
                 ;;
         esac
-        exit $EXIT
         ;;
 
     "cms-unit")
-        paver test_system -s cms --extra_args="--with-flaky" --cov_args="-p" || { EXIT=1; }
-        exit $EXIT
+        paver test_system -s cms --extra_args="--with-flaky" --cov_args="-p"
         ;;
 
     "commonlib-js-unit")
         paver test_js --coverage --skip_clean || { EXIT=1; }
         paver test_lib --skip_clean --extra_args="--with-flaky" --cov_args="-p" || { EXIT=1; }
+
+        # This is to ensure that the build status of the shard is properly set
```

Comment:
I'd say something like: 
Because we are running two paver commands in a row, we need to capture their return codes in order to exit with a non-zero code if either of them fail.
We put the `||` clause there because otherwise when a paver command fails, this entire script will exit, and not run the second paver command in this case statement.
So instead of exiting, the value of a variable named `EXIT` will be set to 1 if either of the paver commands fail.
We then use this variable's value as our exit code.
Note that by default the value of this variable `EXIT` is not set, so if neither command fails then the exit command resolves to simply `exit ` which is considered successful.